### PR TITLE
Delete experiment in DB if in need

### PR DESCRIPTION
**Pull request by hougangliu**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/519)

### CODE REVIEW

File: pkg/api/operators/apis/experiment/v1alpha2/util.go [hunk 1]
```diff
@@ -137,3 +141,33 @@ func (exp *Experiment) MarkExperimentStatusFailed(reason, message string) {
...
+			contained = true
+			break
+		}
+	}
+
+	if !deleted && !contained {
+		if exp.Spec.RetainHistoricalData {
+			return false, []string{}
+		}
+		finalizers := append(pendingFinalizers, cleanDataFinalizer)
```
Comment (YujiOshima) [inline pkg/api/operators/apis/experiment/v1alpha2/util.go]: Why do we need `cleanDataFinalizer` and exclude it at line 166?

Comment (hougangliu) [reply pkg/api/operators/apis/experiment/v1alpha2/util.go]: by k8s [finalizer feature](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers), we can make sure when "kubectl delete experiment xx", the related data in DB deleted before experiment resource deleted in k8s.

Comment (YujiOshima) [reply pkg/api/operators/apis/experiment/v1alpha2/util.go]: I see. Thanks.

**Pull request merged**